js_modules/dagit/packages/core/src/ui/BaseTag.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import styled from 'styled-components/macro';
+
+import {ColorsWIP} from './Colors';
+import {IconWrapper} from './Icon';
+
+interface Props {
+  fillColor?: string;
+  iconColor?: string;
+  textColor?: string;
+  icon?: React.ReactNode;
+  rightIcon?: React.ReactNode;
+  label?: React.ReactNode;
+}
+
+export const BaseTag = (props: Props) => {
+  const {
+    fillColor = ColorsWIP.Gray10,
+    iconColor = ColorsWIP.Gray900,
+    textColor = ColorsWIP.Gray900,
+    icon,
+    rightIcon,
+    label,
+  } = props;
+  return (
+    <StyledTag $fillColor={fillColor} $textColor={textColor} $iconColor={iconColor}>
+      {icon || null}
+      {label ? <span>{label}</span> : null}
+      {rightIcon || null}
+    </StyledTag>
+  );
+};
+
+interface StyledTagProps {
+  $fillColor: string;
+  $textColor: string;
+  $iconColor: string;
+}
+
+const StyledTag = styled.div<StyledTagProps>`
+  background-color: ${({$fillColor}) => $fillColor};
+  border-radius: 8px;
+  color: ${({$textColor}) => $textColor};
+  cursor: default;
+  display: inline-flex;
+  flex-direction: row;
+  font-size: 12px;
+  line-height: 16px;
+  align-items: center;
+  padding: 4px 8px;
+  user-select: none;
+
+  > ${IconWrapper}:first-child {
```

Comment:
I can't quite tell what the end effect is here but it might be a fun use of `gap: 4px`